src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformUtils.java

```diff
@@ -101,10 +103,18 @@ public static Platform getPlatformProto(Spawn spawn, @Nullable RemoteOptions rem
       }
     }
 
+    platformBuilder.addAllProperties(extraProperties);
+
     sortPlatformProperties(platformBuilder);
     return platformBuilder.build();
   }
 
+  @Nullable
+  public static Platform getPlatformProto(Spawn spawn, @Nullable RemoteOptions remoteOptions)
```

Comment:
nit: `getPlatformProto()` only has four call sites, so this overload is probably unnecessary

Reply:
Done. Changed to use value of `DIFFERENTIATE_WORKSPACE_CACHE` instead of `execRoot.getBaseName()`.